analyses/ewings-cell-type/03-copykat.Rmd

```diff
@@ -0,0 +1,433 @@
+
+### Heatmap with endothelial cells as reference
+
+![](`r file.path(results_dir, "endothelial_normal", "SCPCL000822_copykat_heatmap.jpeg")`)
+
+Both of these heatmaps appear to show some increase in observed CNVs in aneuploid vs. diploid which is to be expected. 
+
+### UMAP
+
+Below we prepare and plot a UMAP that shows which cells are classified as diploid, aneuploid, and not defined by `CopyKAT`. 
+We show a side by side UMAP with results from running `CopyKAT` both with and without a reference of normal cells. 
+
+```{r}
+umap_df <- sce |> 
+  scuttle::makePerCellDF(use.dimred = "UMAP") |> 
+  # replace UMAP.1 with UMAP1
+  dplyr::rename_with(
+        \(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")
+      )
+
+cnv_df <- umap_df |> 
+  # first add manual annotations
+  dplyr::left_join(manual_classifications_df) |> 
+  # now add copykat results
+  dplyr::left_join(ck_results_df, by = c("barcodes" = "cell.names"))
+```
+
+```{r}
+ggplot(cnv_df, aes(x = UMAP1, y = UMAP2, color = copykat.pred)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() +
+  facet_wrap(vars(reference_used))
+```
+
+If we look at the UMAPs shown in `01-marker-gene-tumor-classification.Rmd`, the cells that are labeled aneuploid here are consistent with where we saw marker gene expession for tumor cells. 
+It also looks like there is not a visible difference between using a reference of normal cells and not using a reference. 
+
+### Validate common CNAs found in Ewing sarcoma
+
+To validate some of these annotations, we can also look at some commonly found copy number variations found in Ewing sarcoma patients. 
+There are a few known copy number variations in Ewing's sarcoma: 
+
+- Gain of Chr8
+- Gain of Chr12
+- Gain of Chr1p
+- Loss of Chr16q
+
+Although these are the most frequent, there are patients who do not have any of these alterations and patients that only have some of these alterations. 
+See [Tirode et al.,](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4264969/) and [Crompton et al.](https://doi.org/10.1158/2159-8290.CD-13-1037). 
+
+`CopyKAT` outputs a matrix that contains the estimated copy numbers for each gene in each cell. 
+We can read that in and look at the mean estimated copy numbers for each chromosome across each cell. 
+We might expect that tumor cells would show an increased estimated copy number in Chr8, Chr12, and/or Chf 1 and a loss of Chr16. 
+
+```{r}
+# for every cell, calculate the mean detection level across all genes in a given chromosome
+full_cnv_df <- full_ck_results_df |> 
+  tidyr::pivot_longer(cols = -c(
+    reference_used,
+    abspos, 
+    chromosome_name, 
+    start_position, 
+    end_position, 
+    ensembl_gene_id, 
+    hgnc_symbol, 
+    band),
+    names_to = "barcodes",
+    values_to = "cnv_detection") |> 
+  dplyr::group_by(chromosome_name, barcodes, reference_used) |> 
+  dplyr::summarise(mean_cnv_detection = mean(cnv_detection)) 
```

Comment:
> Is there any stat for the significance of the cnv_detection? I feel like what we would want here is not the average over all genes but something that takes into account that some are going to be more informative than others.

I looked for stats on the calls and have found nothing... there's really not a lot in the output. This is why I'm hesitant to really interpret these plots too much since all I have is this arbitrary number for detection. Again, there is no "predictions" table, just labeling of each cell as aneuploid and diploid.